java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestGenerationUtils.java

```diff
@@ -152,16 +163,280 @@ public static WorkspaceEdit generateTests(List<Object> arguments, IProgressMonit
+            TestKind testKind, String delimiter) throws CoreException {
+        final String typeName = testUnit.getElementName().replace(".java", "");
+        final StringBuilder buf = new StringBuilder();
+        buf.append("public class ").append(typeName).append(" {").append(delimiter);
+        for (final String method : methods) {
+            buf.append(constructMethodStub(testUnit, testKind, method, delimiter)).append(delimiter);
+        }
+        buf.append("}").append(delimiter);
+        return buf.toString();
+    }
+
+    private static String constructMethodStub(ICompilationUnit testUnit, TestKind testKind,
+            String method, String delimiter) {
+        final StringBuilder buf = new StringBuilder();
+        buf.append("@Test").append(delimiter);
+        if (testKind == TestKind.JUnit) {
+            buf.append("public ");
+        }
+        final String methodName = getTestMethodName(method);
+        buf.append("void ").append(methodName).append("() {").append(delimiter).append(delimiter).append("}");
+        final String methodContent = buf.toString();
+        // TODO: get test unit options directly
+        return CodeFormatterUtil.format(CodeFormatter.K_STATEMENTS, methodContent, 1,
+                delimiter, testUnit.getJavaProject().getOptions(true));
+    }
+
+    private static IClasspathEntry getTestClasspathEntry(ICompilationUnit unit) throws JavaModelException {
+        final IJavaProject javaProject = unit.getJavaProject();
+        // In most cases, this is the classpath entry used for testing, we first find the target entry by hard-code
+        // to avoid go into the generated entries.
+        final IClasspathEntry testEntry = javaProject.getClasspathEntryFor(
+            javaProject.getPath().append("src/test/java"));
+        if (ProjectTestUtils.isTestEntry(testEntry)) {
+            return testEntry;
+        }
+
+        final IClasspathEntry[] entries = javaProject.readRawClasspath();
+        for (final IClasspathEntry entry : entries) {
+            if (ProjectTestUtils.isTestEntry(entry)) {
+                return entry;
+            }
+        }
+
+        return entries[0];
+    }
+
+    private static String getTestFullyQualifiedName(ITypeBinding typeBinding, IJavaProject project,
+            IClasspathEntry testEntry) throws JavaModelException {
+        String promptName = typeBinding.getBinaryName() + "Tests";
+        final ICompilationUnit testCompilationUnit = getTestCompilationUnit(project, testEntry, promptName);
+        if (!testCompilationUnit.exists()) {
+            promptName = typeBinding.getBinaryName() + "Test";
+        }
+
+        final String fullyQualifiedName = (String) JUnitPlugin.askClientForInput(
+            "Please type the target test class name", promptName);
+        if (fullyQualifiedName == null) {
+            return null;
+        }
+
+        if (fullyQualifiedName.charAt(fullyQualifiedName.length() - 1) == '.') {
```

Comment:
what's the case?

Reply:
Got it.  If it was me, I prefer not to treat this as a special case. My proposal:
* split with `.`
* check each part, call it `chars`, if any char is invalid or `chars` is empty, report error.

Reply:
It's because split in Java will ignore the trailing empty string. For example, `"a.b.c.".split("\\.")` will result to `["a", "b", "c"]`

Reply:
got it